dotnet/src/webdriver/Alert.cs

```diff
@@ -46,7 +48,7 @@ public string Text
             get
             {
                 Response commandResponse = this.driver.InternalExecute(DriverCommand.GetAlertText, null);
-                return commandResponse.Value.ToString();
+                return (string)commandResponse.Value!;
```

Comment:
The [spec](https://w3c.github.io/webdriver/#get-alert-text) says:

1. If session's [current top-level browsing context](https://w3c.github.io/webdriver/#dfn-current-top-level-browsing-context) is [no longer open](https://w3c.github.io/webdriver/#dfn-no-longer-open), return [error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [no such window](https://w3c.github.io/webdriver/#dfn-no-such-window).

2. If the [current user prompt](https://w3c.github.io/webdriver/#dfn-current-user-prompt) is null, return [error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [no such alert](https://w3c.github.io/webdriver/#dfn-no-such-alert).

3. Let message be the text message associated with the [current user prompt](https://w3c.github.io/webdriver/#dfn-current-user-prompt), or otherwise be [null](https://w3c.github.io/webdriver/#dfn-null).

4. Return [success](https://w3c.github.io/webdriver/#dfn-success) with data message.

It's a little unclear to me when `message` is set to `null`. I tried as hard as I could with all kinds of edge cases, but I couldn't get it to return anything besides `string.Empty`. Since a `null` would have caused a null-reference here, I'm going to assume it cannot happen, so I'm marking this as non-nullable return.

Reply:
I am OK to keep it as non-nullable at declaration level. I guess the easiest way to verify it is to execute JS: `alert()`. If the `Text` is string `string.Empty` then we are "safe".

About `.ToString()` vs `(string)`: please think about what exception user will see in case of `value is null`.

Reply:
`InvalidCast` is better than `NullReference`, right.

Reply:
I tried with all the combinations, from `alert()` to `prompt()`. It always returns `string.Empty`.

About the exceptions, if we get `value is null` then the current way will throw `NullReferenceException` on the `.ToString()` call, while my proposed way will just return `null`. That's why I'm pretty confident this doesn't return null: we would have gotten bug reports about it.

If we want to be completely sure, we can do `Value ?? throw new WebDriverException("Alert text returned null")`. But maybe that's not necessary.

Reply:
done